bodhi/server/scripts/skopeo_lite.py

```diff
@@ -0,0 +1,472 @@
+            for layer in manifest['layers']:
+                references.append((layer['digest'], layer['size']))
+        else:
+            raise RuntimeError("Unhandled media type %s", info.media_type)
+
+        for digest, size in references:
+            self._copy_blob(digest, size)
+
+        self.dest.write_manifest(info, toplevel=toplevel)
+
+    def copy(self):
+        self.dest.start()
+        info = self.src.get_manifest()
+        if info.media_type in (MEDIA_TYPE_MANIFEST_V2, MEDIA_TYPE_OCI):
+            self._copy_manifest(info, toplevel=True)
+        elif info.media_type in (MEDIA_TYPE_LIST_V2, MEDIA_TYPE_OCI_INDEX):
+            manifest = json.loads(info.contents)
+            for m in manifest['manifests']:
+                referenced = self.src.get_manifest(digest=m['digest'], media_type=m['mediaType'])
+                self._copy_manifest(referenced)
+            self.dest.write_manifest(info, toplevel=True)
+
+        else:
+            raise RuntimeError("Unhandled media type %s", info.media_type)
+
+
+@main.command()
+@click.option('--src-creds', '--screds', metavar='USERNAME[:PASSWORD]',
+              help='Use USERNAME[:PASSWORD] for accessing the source registry')
+@click.option('--src-tls-verify', type=bool, default=True,
+              help='require HTTPS and verify certificates when talking to the ' +
+              'container source registry (defaults to true)')
+@click.option('--src-cert-dir', metavar='PATH',
+              help='use certificates at PATH (*.crt, *.cert, *.key) to connect to the ' +
+              'source registry')
+@click.option('--dest-creds', '--dcreds', metavar='USERNAME[:PASSWORD]',
+              help='Use USERNAME[:PASSWORD] for accessing the destination registry')
+@click.option('--dest-tls-verify', type=bool, default=True,
+              help='require HTTPS and verify certificates when talking to the ' +
+              'container destination registry (defaults to true)')
+@click.option('--dest-cert-dir', metavar='PATH',
+              help='use certificates at PATH (*.crt, *.cert, *.key) to connect to the ' +
+              'destination registry')
+@click.argument('src', metavar='SOURCE-IMAGE-NAME')
+@click.argument('dest', metavar='DEST-IMAGE-NAME')
+def copy(src, dest, screds, src_tls_verify, src_cert_dir, dcreds, dest_tls_verify, dest_cert_dir):
+    """Copy an image from one location to another."""
+    src = parse_spec(src, screds, src_tls_verify, src_cert_dir)
+    dest = parse_spec(dest, dcreds, dest_tls_verify, dest_cert_dir)
+
+    if src.registry != dest.registry:
+        tempdir = tempfile.mkdtemp()
+        try:
+            tmp = DirectoryEndpoint(tempdir)
+            Copier(src.get_endpoint(), tmp).copy()
+            Copier(tmp, dest.get_endpoint()).copy()
+        finally:
+            shutil.rmtree(tempdir)
+    else:
+        Copier(src.get_endpoint(), dest.get_endpoint()).copy()
+
+
+if __name__ == '__main__':
+    main()
+
+__all__ = ['copy', 'main']
```

Comment:
Bodhi doesn't use ```import *``` and I am morally opposed to the use of ```import *```, so we don't need this here. IMO, it's better to use leading underscores if the intention is to indicate private vs. public methods.

Reply:
My goal here was to avoid writing doc strings for a lot of classes/methods internal to the file - to make pydocstyle ignore them - with the idea that this is a self-contained script - I can _ prefix them - but it sounds from what you say elsewhere you'd prefer even internal functions have dockblocks.